pkg/parser/extended_syntax_parser.y

```diff
@@ -158,6 +166,12 @@ sqlflow_select_stmt
 		Train: true,
 		TrainClause: $1}
   }
+| show_train_clause end_of_stmt {
+	parseResult = &SQLFlowSelectStmt{
+		Extended: true,
+		ShowTrain: true,
+		ShowTrainClause: $1}
+}
```

Comment:
Adding the SHOW TRAIN clause here would enable users to write 

```sql
SELECT ... SHOW TRAIN trained_model_name;
```

However, according to the design doc https://github.com/sql-machine-learning/sqlflow/pull/2073/files#diff-49e684e27ee27bc9f191167dcd303674R28, we don't need the SELECT ... prefix.

```sql
SHOW TRAIN trained_model_name;
```

How are we going to have it? Is that the realization of the idea depends on the improvement of the collaborative parsing algorithm as the pseudo-code shows: https://github.com/sql-machine-learning/sqlflow/pull/2073/files#diff-49e684e27ee27bc9f191167dcd303674R46

Reply:
This is the first part of our modification, we just add syntax rule here. In the second part, we will change the behavior of collaborative parsing to exlude this case.